src/util.js

```diff
@@ -835,3 +835,32 @@ export function generateMobileDialogEl() {
 
     return el;
 }
+
+/**
+ * Calculates the dialog width if the highlighter's name is to be displayed
+ * in the annotations dialog
+ *
+ * @param {HTMLElement} dialogEl - Annotation dialog element
+ * @return {number} Annotations dialog width
+ */
+export function getDialogWidth(dialogEl) {
```

Comment:
I see that it was like this before, but is there a reason for hiding/showing the dialog so many times in this function? There may be a lot of layout recalculation going on when making these calls, and even the possibility of incorrect values being returned

Reply:
Hmm so I think the reason this was done this way was so that we could get the entire dialog width and accurately position the dialog before we actually display it. Then we wouldn't have the dialog shown and THEN repositioned correctly. 

I could definitely see how it could cause some recalculations. I am open to suggestions on how to optimize this :)

Reply:
@JustinHoldstock any thoughts on how to improve this?